### PR TITLE
Add support for sign

### LINKED ISSUE #44: `sign` not supported
```vim
sign define DiagnosticSignError text=✖ texthl=DiagnosticSignError linehl= numhl=
sign define DiagnosticSignWarn text=⚠ texthl=DiagnosticSignWarn linehl= numhl=
sign define DiagnosticSignInfo text=ℹ texthl=DiagnosticSignInfo linehl= numhl=
sign define DiagnosticSignHint text=💡 texthl=DiagnosticSignHint linehl= numhl=
```

```
ERROR [254, 0] - [257, 81]
  identifier [254, 0] - [254, 4]
  identifier [254, 5] - [254, 11]
  identifier [254, 12] - [254, 31]
  identifier [254, 32] - [254, 36]
  ERROR [254, 37] - [254, 40]
  identifier [254, 41] - [254, 47]
  identifier [254, 48] - [254, 67]
  identifier [254, 68] - [254, 74]
  identifier [254, 76] - [254, 81]
  identifier [255, 0] - [255, 4]
  identifier [255, 5] - [255, 11]
  identifier [255, 12] - [255, 30]
  identifier [255, 31] - [255, 35]
  ERROR [255, 36] - [255, 39]
  identifier [255, 40] - [255, 46]
  identifier [255, 47] - [255, 65]
  identifier [255, 66] - [255, 72]
  identifier [255, 74] - [255, 79]
  identifier [256, 0] - [256, 4]
  identifier [256, 5] - [256, 11]
  identifier [256, 12] - [256, 30]
  identifier [256, 31] - [256, 35]
  ERROR [256, 36] - [256, 39]
  identifier [256, 40] - [256, 46]
  identifier [256, 47] - [256, 65]
  identifier [256, 66] - [256, 72]
  identifier [256, 74] - [256, 79]
  identifier [257, 0] - [257, 4]
  identifier [257, 5] - [257, 11]
  identifier [257, 12] - [257, 30]
  identifier [257, 31] - [257, 35]
  ERROR [257, 36] - [257, 40]
  identifier [257, 41] - [257, 47]
  identifier [257, 48] - [257, 66]
  identifier [257, 67] - [257, 73]
  identifier [257, 75] - [257, 80]
```

### PR DESCRIPTION
**Pull request by Fymyte**
closes #44
Signed-off-by: Fymyte <pierguill@gmail.com>

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -854,6 +858,126 @@ module.exports = grammar({
...
+
+    _sign_define_arg_text: ($) => seq($._printable, optional($._printable)),
+    _sign_define_argument: ($) =>
+      choice(
+        // FIXME: use filename node when filename is merged
+        key_val_arg('icon', optional($.filename)),
+        key_val_arg('linehl', optional($.hl_group)),
+        key_val_arg('numhl', optional($.hl_group)),
+        key_val_arg('text', optional(alias($._sign_define_arg_text, $.text))),
+        key_val_arg('texthl', optional($.hl_group)),
```
Comment (vigoux) [inline grammar.js:873]: There is `culhl` missing here.

**Review by vigoux — APPROVED** [review]
Great ! The refactoring is also very nice to see !
Great job !

File: grammar.js (hunk 1 above)
Comment (Fymyte) [reply grammar.js:873]: I don't have culhl in my nvim help, is this a vim specific or nvim-v7 ?

Comment (Fymyte) [reply grammar.js:872]: Yep found it in the neovim doc on github :+1:

**Pull request merged**